src/sentry/workflow_engine/endpoints/validators/base/workflow.py

```diff
@@ -1,22 +1,103 @@
+from typing import Any
+
+from django.db import router, transaction
 from rest_framework import serializers
 
+from sentry import audit_log
 from sentry.api.serializers.rest_framework import CamelSnakeSerializer
-from sentry.workflow_engine.endpoints.validators.base import BaseDataConditionGroupValidator
+from sentry.utils.audit import create_audit_entry
+from sentry.workflow_engine.endpoints.validators.base import (
+    BaseActionValidator,
+    BaseDataConditionGroupValidator,
+)
 from sentry.workflow_engine.endpoints.validators.utils import validate_json_schema
-from sentry.workflow_engine.models import Workflow
+from sentry.workflow_engine.models import (
+    DataConditionGroupAction,
+    Workflow,
+    WorkflowDataConditionGroup,
+)
+
+DataConditionGroupData = dict[str, Any]
+ActionData = dict[str, Any]
 
 
 class WorkflowValidator(CamelSnakeSerializer):
     name = serializers.CharField(required=True, max_length=256)
     enabled = serializers.BooleanField(required=False, default=True)
     config = serializers.JSONField(required=False)
     triggers = BaseDataConditionGroupValidator(required=False)
-    action_filters = serializers.ListField(child=BaseDataConditionGroupValidator(), required=False)
+    action_filters = serializers.ListField(required=False)
 
     # TODO - Need to improve the following fields (validate them in db)
     organization_id = serializers.IntegerField(required=True)
     environment_id = serializers.IntegerField(required=False)
 
+    def _split_action_and_condition_group(
+        self, action_filter: dict[str, Any]
+    ) -> tuple[ActionData, DataConditionGroupData]:
+        action = action_filter.get("action", None)
```

Comment:
should this account for more than 1 action because more than 1 action can be attached to a `DataConditionGroup`?